pkg/controller/core/localqueue_controller.go

```diff
@@ -63,8 +63,13 @@ func NewLocalQueueReconciler(client client.Client, queues *queue.Manager, cache
 	}
 }
 
-func (r *LocalQueueReconciler) NotifyWorkloadUpdate(w *kueue.Workload) {
-	r.wlUpdateCh <- event.GenericEvent{Object: w}
+func (r *LocalQueueReconciler) NotifyWorkloadUpdate(oldWl, newWl *kueue.Workload) {
+	if oldWl != nil {
+		r.wlUpdateCh <- event.GenericEvent{Object: oldWl}
+	}
+	if newWl != nil {
```

Comment:
also check if the local queue changed, otherwise there is no need to add to updateCh again.